### PR TITLE
[Cache] Prevent fatal errors on php 8 when running concurrently with TagAwareAdapter v6.1

### PR DESCRIPTION
**Pull request by sbelyshkin**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 - 5.3
| Bug fix?      | no
| New feature?  |  no 
| Deprecations? |  no 
| Tickets       | - 
| License       | MIT
| Doc PR        |

This fix prevents fatal errors which may arise during rolling updates when TagAwareAdapter is upgraded from versions prior to 5.4 to the latest 6.1. TypeErrors occur because of different value types for tag versions and rules for numeric strings in php 8.x https://www.php.net/manual/en/language.types.numeric-strings.php

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php [hunk 1]
```diff
@@ -177,9 +177,10 @@ public function hasItem($key)
         }
 
         foreach ($this->getTagVersions([$itemTags]) as $tag => $version) {
-            if ($itemTags[$tag] !== $version && 1 !== $itemTags[$tag] - $version) {
-                return false;
+            if ($itemTags[$tag] === $version || \is_int($itemTags[$tag] && \is_int($version) && 1 === $itemTags[$tag] - $version)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php]: `is_int($itemTags[$tag]` is missing its ending bracket

Comment (sbelyshkin) [reply src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php]: Thank you @nicolas-grekas!

File: src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php [hunk 2]
```diff
@@ -430,6 +433,10 @@ private function getTagVersions(array $tagsByKey, array &$invalidatedTags = [])
             if (isset($invalidatedTags[$tag])) {
                 $invalidatedTags[$tag] = $version->set(++$tagVersions[$tag]);
             }
+            if (!\is_int($tagVersions[$tag])) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php:436]: on the line above, we have a `++` on this entry, shouldn't we check for this earlier?

Comment (sbelyshkin) [reply src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php:436]: ++ operator works with strings
Although it not always changes string values (and thus no invalidation occurs in those cases) but at least it doesn't do the fatality.

**Comment by xabbuh** [comment]
Can we add a test to prevent future regressions?

**Comment by sbelyshkin** [comment]
I've added a few tests. Please let me know if they make sense.

**Comment by nicolas-grekas** [comment]
Thank you @sbelyshkin.

**Pull request merged**